### PR TITLE
Fix seperation of unknown_tokens

### PR DESCRIPTION
**Pull request by na-g**
Nils, it looks like the `red`, `dtmf`, and `unknown_parameters` were being concatenated incorrectly on the FTMP parameters.

### CODE REVIEW

File: src/attribute_type.rs [hunk 1]
```diff
@@ -652,12 +652,12 @@ impl fmt::Display for SdpAttributeFmtpParameters {
...
-                maybe_print_bool_param("cbr", self.cbr, false)
+                maybe_print_bool_param("cbr", self.cbr, false),
+                maybe_vector_to_string!("{}", self.encodings, "/"),
+                maybe_print_param("", self.dtmf_tones.clone(), "".to_string()),
+                maybe_vector_to_string!("{}", self.unknown_tokens, ",")
             ]
-            .join(";"),
-            red = maybe_vector_to_string!("{}", self.encodings, "/"),
-            dtmf = maybe_print_param("", self.dtmf_tones.clone(), "".to_string()),
-            unknown = maybe_vector_to_string!("{}", self.unknown_tokens, ",")
```
Comment (nils-ohlmeier) [inline src/attribute_type.rs:660]: I noticed this too. But I was not sure if there are any spec reasons for keeping these separate. Did you have a chance to check RFCs that this doesn't break something?
It looks like red and dtmf are special in that they don't follow the usual name=value convention.

File: src/attribute_type.rs [hunk 2]
```diff
@@ -4410,4 +4410,36 @@ mod tests {
...
+            encodings: Vec::new(),
+            dtmf_tones: "".to_string(),
+            rtx: None,
+            unknown_tokens: vec!["sprop-parameter-sets=Z0LAFYyNQKD5APCIRqA=,aM48gA==".to_string()],
+        };
+        assert_eq!(
+            format!("{}", att),
+            "packetization-mode=1;sprop-parameter-sets=Z0LAFYyNQKD5APCIRqA=,aM48gA=="
+        );
+    }
```
Comment (nils-ohlmeier) [inline src/attribute_type.rs:4444]: I would propose to make this simpler with something like this:

`check_parse_and_serialize("fmtp:102 packetization-mode=1;profile-level-id=42c015;sprop-parameter-sets=Z0LAFYyNQKD5APCIRqA=,aM48gA==");`

But I would like to see more of these including combinations with red and dtmf with other parameters to make sure we are not screwing up these.

**Review by nils-ohlmeier — CHANGES_REQUESTED** [review]
Let's make the unit test a little easier please.

File: src/attribute_type.rs (hunk 1 above)
Comment (na-g) [reply src/attribute_type.rs:660]: We should split this struct into an enum, which would require a version bump. If it is alright with you, I think we should just fix the missing semi-colon in this patch.

Comment (nils-ohlmeier) [reply src/attribute_type.rs:660]: After you found out that dmtf and red can't be combined with other params, does still make sense to move them into the comma separated list of params?
I would have thought the most straight forward patch would be to only put the unknown_tokens into the semicolon list, but leave dtmf and red outside. And take care of them when we break them out into an enum like you suggested.

**Comment by na-g** [comment]
@nils-ohlmeier, this should be ready for another look.

**Review by nils-ohlmeier — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**